docs/maintenance/internals/jobs.md

```diff
@@ -0,0 +1,182 @@
+# Job System Design
+
+_Added in v13.0.0_
+
+The InvenioRDM job system introduces a robust, extensible mechanism for running and managing asynchronous background tasks. Built on top of Celery, it provides administrators with full visibility, auditability, and control over long-running or scheduled tasks via the admin UI or REST API.
+
+---
+
+## Why a Job System?
+
+Traditionally, Celery tasks in InvenioRDM were launched programmatically or via CLI. This created challenges:
+
+- No standardized logging or audit trail.
+- Hard to expose jobs to administrators.
+- No unified way to track execution history or outcomes.
+- No parameterized job execution from the UI.
+
+The job system addresses these problems by introducing two core concepts:
+
+- **Jobs**: registered task definitions with metadata and optional schemas.
+- **Job Runs**: execution instances of a job, which are persisted and tracked.
+
+---
+
+## Architecture Overview
+
+### Core Components
+
+- **JobType (Base Class)**:
+  Defines the interface for a job: `id`, `title`, `description`, `task`, `arguments_schema`, and `build_task_arguments`.
+
+- **Celery Tasks**:
+  Implement the actual logic. Must be registered as entry points and referenced by the JobType.
+
+- **Job Registry**:
+  Discovers jobs via entry points (`invenio_jobs.jobs`).
+
+- **Job Runs**:
+  Persisted database entries that track execution, status, timestamps, logs, and arguments.
+
+- **Admin UI**:
+  Provides a scheduling form, status view, and log viewer for each job.
+
+- **REST API**:
+  Enables programmatic triggering, listing, and status inspection of jobs and runs.
+
+---
+
+### Scheduler Beat
+
+Scheduled and on-demand job execution requires a **custom scheduler beat**, which is expected to be implemented and deployed by the user. This beat is responsible for dispatching job runs, whether triggered manually or on a schedule.
```

Comment:
```suggestion
Scheduled and on-demand job execution requires a **custom scheduler beat**, which is expected to be implemented and deployed by the instance manager/developer. This beat is responsible for dispatching job runs, whether triggered manually or on a schedule.
```
just because we usually talk about the user as the person who submits, aka end user, but this is not the case here